### ISSUE TITLE
Support debug mode or logging SQL?

### ISSUE THREAD

**Issue by KingWu** [tekartik/sqflite#145]
Any ways to log the insert, update, delete & query SQL statement when operating the statement?

Thanks

**Comment by alextekartik** [comment]
Yes `Sqflite.devSetDebugModeOn` : <https://github.com/tekartik/sqflite/blob/master/doc/dev_tips.md#turn-on-sql-console-logging>

**Comment by KingWu** [comment]
i turn on the debug mode. but the logging is not helpful. i cannot find the sql statement in the logs

**Comment by KingWu** [comment]
The log shows in unicode format

**Comment by alextekartik** [comment]
I'm not sure it plays well with unicode though and I have seen many cases where sometimes the logs got stuck on Android and not appearing on iOS. I don't have better solution at the moment. You could try to log the command on your side to see if you get better result

**Comment by jaumard** [comment]
@alextekartik would be nice to have a way to give a custom logger to Sqflite like `Sqflite.setDebugModeOn(true, Logger('SQLITE'))`, it would allow people to also retrieve sql logs and do whatever they want with it (sent it over crashlytics if there a crash for example).
Or if you want to be logger agnostic something like `Sqflite.setDebugModeOn(true, (message) { Logger('SQLITE').info(message)})`

**Comment by jaumard** [comment]
I also agree with @KingWu that SQL raw query should be on the logs when debug mode is activated, we only get the result rows

**Comment by givip** [comment]
`Sqflite.devSetDebugModeOn(true);` not working at all on sqflite 1.3.1. Is there legal way to log sql queries?

**Comment by givip** [comment]
Comment mentioned above is valid for iOS, on Android emulator SQL queries are visible in console.

**Comment by danypata** [comment]
Comments work fine on Android, but on iOS won't work due to general logging issue with print/NSLog. But if you still want to see the logs, then you can use the ```console``` app on OSX.

**Comment by alextekartik** [comment]
https://pub.dev/documentation/sqflite_common/latest/sqflite_logger/

**Comment by sbrsubuvga** [comment]
[sqflite_dev](https://pub.dev/packages/sqflite_dev)
A developer dependency package that provides a web-based SQLite workbench for Flutter and pure Dart apps during development. Works with both sqflite (mobile) and sqflite_common_ffi (desktop).
